lib/rhc/ssh_helpers.rb

```diff
@@ -490,9 +490,11 @@ def check_ssh_executable!(path)
           raise RHC::InvalidSSHExecutableException.new("No system SSH available. Please use the --ssh option to specify the path to your SSH executable, or install SSH.#{windows? ? ' We recommend this free application: Git for Windows - a basic git command line and GUI client http://msysgit.github.io/.' : ''}") unless ssh_cmd or has_ssh?
         end
       else
-        bin_path = split_path(path)[0]
-        raise RHC::InvalidSSHExecutableException.new("SSH executable '#{bin_path}' does not exist.") unless File.exist?(bin_path) or File.exist?(path) or exe?(bin_path)
-        raise RHC::InvalidSSHExecutableException.new("SSH executable '#{bin_path}' is not executable.") unless File.executable?(path) or File.executable?(bin_path) or exe?(bin_path)
+        if File.file?(path)
+          raise RHC::InvalidSSHExecutableException.new("SSH executable '#{path}' does not exist.") if (File.exist?(path) and File.readable?(path)) or exe?(path)
+        else
+          raise RHC::InvalidSSHExecutableException.new("The path '#{path}' is not a file. Please provide a SSH executable file instead.")
+        end
```

Comment:
We want `check_ssh_executable!` to return `path` if it is the full path to an executable and regular file, but we also want to return the path if it is the name of an executable that `exe?` can find in `$PATH`.  The situation we need to check for, that `check_ssh_executable!` does not already check for, is where a full path to an executable is provided but that executable is a directory rather than a regular file.  (While we are at it, we might as well exclude device files, named pipes, and any other special files.) 

I think it would be better to keep the existing lines and add a check along the lines of `raise RHC::InvalidSSHExecutableException.new("SSH executable '#{bin_path}' is not a regular file") if File.exist?(bin_path) and !File.file?(bin_path)`.  We should also modify the `exe?` method so that it checks `File.file?` in addition to (or instead of) `File.exist?` and in addition to `File.executable?`.